packages/affiliates/libs/affiliates.js

```diff
@@ -91,52 +92,60 @@ const DeployerRewards = ({ queries, empCreatorAbi, empAbi, coingecko }) => {
 
+    const rewardsPerBlock = toBN(toWei(totalRewards)).div(toBN(blocks.length));
+    const payoutPerSnapshot = rewardsPerBlock.mul(toBN(snapshotSteps));
     const valuePerSnapshot = blocks.reduce((result, block, index) => {
       if (index % snapshotSteps !== 0) return result;
       const { timestamp } = block;
-
       const valueByEmp = empWhitelist.reduce((result, empAddress, empIndex) => {
         try {
           const { tokens } = balanceHistories.get(empAddress).history.lookup(block.number);
           const decimals = tokenDecimals[empIndex];
           assert(decimals, "requires token decimals lookup");
           const [, closestPrice] = tokenPrices[empIndex].closest(timestamp);
-          const totalTokens = Object.values(tokens).reduce((a, b) => a + BigInt(b), 0n);
-          // console.log({closestPrice,totalTokens},BigInt(parseFixed(closestPrice.toFixed(18),18)))
-          // need to conver total tokens to consistent decimals across all tokens
+          const totalTokens = Object.values(tokens).reduce((result, value) => {
+            return result.add(toBN(value));
+          }, toBN("0"));
           result.push([empAddress, calculateValue(totalTokens, closestPrice, decimals)]);
           return result;
         } catch (err) {
-          // this error is ok, it means we have block history before the emp had
-          // any events. this essentially means value locked at emp is 0 at this block.
+          // this error is ok, it means we have block history before the emp had any events. Locked value is 0 at this block.
           result.push([empAddress, 0]);
           return result;
         }
       }, []);
-
       result.push(valueByEmp);
       return result;
     }, []);
 
-    // per snapshot
-    return valuePerSnapshot.reduce((result, valueByEmp) => {
+    // Per snapshot calculate the associated amount that each deployer is entitled to.
+    let finalDeployerPayouts = valuePerSnapshot.reduce((result, valueByEmp) => {
       const totalValueLocked = valueByEmp.reduce((result, [, value]) => {
-        return result + BigInt(value);
-      }, 0n);
+        return result.add(toBN(value));
+      }, toBN("0"));
       valueByEmp.forEach(([emp, value]) => {
-        // console.log({value,totalValueLocked})
         const deployer = empDeployers.get(emp);
-        // this math needs work
-        const contribution = totalValueLocked > 0n ? (BigInt(value) * (2n * 10n ** 18n)) / totalValueLocked : 0n;
-        const rewards = contribution * payoutPerSnapshot;
-        if (result[deployer] == null) result[deployer] = 0n;
-        result[deployer] = result[deployer] + rewards;
+        const contribution =
+          totalValueLocked.toString() != "0"
+            ? toBN(value) // eslint-disable-line indent
+                .mul(toBN(toWei("1"))) // eslint-disable-line indent
+                .div(totalValueLocked) // eslint-disable-line indent
+            : toBN("0"); // eslint-disable-line indent
```

Comment:
just curious -- why do we need all these eslint disables?

Reply:
prettier formats it like this for long length ternary operators. eslint doesn't like it!